uwblueprint/childrens-aid-society
# [kai/intake-post-route] Implement /intake routes and intake related services

## Notion ticket link

[Implement POST /intake route](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)




## Implementation description
* Using the json input, create the necessary dependencies that an `intake` needs (e.g. `caregiver`s, `child`s, etc.) then create the `intake`
* If any dependency fails to be created (e.g. invalid fields) then roll back everything

* I am not 100% sure which fields in the JSON correspond to which models and which models' fields so I did some guessing as to what maps to what. Also, some mandatory fields I'm not sure about where they are in the JSON so I hardcoded stuff (e.g. email="test@mail.com") and also left some question mark comments in `routes/intake.py`



## Steps to test
Before continuing, please run:
```bash
docker-compose down --volumes
docker-compose up -d
docker exec -it cas_py_backend /bin/bash -c "flask db upgrade"
docker exec -it cas_py_backend /bin/bash -c "python -m tools.db_seed"
```

1. When POSTing with a correct payload, the intake and all relevant foreign dependencies are created (check that they exist in the db). One possible way to test is running `good.sh` (but this only checks the `intakes` db)
2. When POSTing with an incorrect payload (e.g. a mandatory field was left blank), not a single change is made to any table (check this as well). One possible way to test is running `bad.sh` (but this only checks the `intakes` db)



## What should reviewers focus on?
* 


## Checklist
